app/src/main/java/org/mozilla/focus/utils/RemoteKey.kt

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.utils
+
+import android.view.KeyEvent
+
+/** Keys the user can press on the remote. */
+enum class RemoteKey {
+    UP, DOWN, LEFT, RIGHT,
+    CENTER;
+
+    fun toDirection() = when (this) {
+        UP -> Direction.UP
+        DOWN -> Direction.DOWN
+        LEFT -> Direction.LEFT
+        RIGHT -> Direction.RIGHT
+        else -> null
+    }
+
+    companion object {
+        fun fromKeyEvent(event: KeyEvent) = when (event.keyCode) {
+            KeyEvent.KEYCODE_DPAD_UP -> UP
```

Comment:
I assume you mean why don't I just switch on KeyEvent.keyCode, rather than converting it to an enum and switching on that.

Using an enum is
- a little less error-prone (e.g. `when` statements for enums must be exhaustive, can't be added to, and are under our control, making it easier to not use a wrong value, like if KEY_ENTER and KEY_RETURN both exist)
- Allows a new type, for convenience functionality (e.g. converting a RemoteKey -> Direction in my PR)

It's entirely possible I'm overdoing the abstraction layer here though... :\ I'll think about it for my next iteration (especially because I'm trying to provide a good baseline perf for this).

Reply:
fwiw, maybe it'd be a better abstraction if I called `dispatchKeyEvent(RemoteKey)` to our internal classes, rather than having each view come up with its own way of handling key events (though this doesn't take into account the keyboard case).